### PR TITLE
Minor refactoring and better handling for inverters that are offline

### PR DESCRIPTION
**Pull request by CM000n**
Hello, I took a look at the code for APsystems-EZ1-API today and have a few minor suggestions for refactoring and code simplification.

I have also implemented better handling of inverters that are offline. Instead of errors, none values are now returned in these cases. This may also fix https://github.com/SonnenladenGmbH/APsystems-EZ1-API-HomeAssistant/issues/8

I also noticed that there are currently no unit tests or anything similar. If there is interest, I could take care of this in another PR.

### CODE REVIEW

File: APsystemsEZ1/__init__.py [hunk 1]
```diff
@@ -234,9 +234,10 @@ async def get_device_power_status(self) -> Status:
 
         :return: 0/normal when on, 1/alarm when off
         """
-        return Status(int((await self._request("getOnOff"))["data"]["status"]))
+        response = await self._request("getOnOff")
+        return Status(response["data"]["status"]) if response else None
```
Comment (mawoka-myblock) [inline APsystemsEZ1/__init__.py:238]: Isn't here a int-mapping missing? So that the enum works?

File: README.md [hunk 2]
```diff
@@ -63,19 +71,23 @@ The local API access needs to be activated once in the settings of the EZ1. Plea
 </ul>
 
 ---
+
 ## Installation
+
 - To use the APsystemsEZ1 library, you need to have Python >=3.8 installed on your system.
-- See the following guide to install the latest Python release: https://www.python.org/downloads <br><br>
+- See the following guide to install the latest Python release: <https://www.python.org/downloads> <br><br>
```
Comment (mawoka-myblock) [inline README.md:78]: Why did you put the Links into the `<` and `>`?

**Review by mawoka-myblock — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (CM000n) [reply README.md:78]: This ist following `MD034` rule for markdown linting: https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md034---bare-url-used

File: APsystemsEZ1/__init__.py (hunk 1 above)
Comment (CM000n) [reply APsystemsEZ1/__init__.py:240]: fixed

**Comment by mawoka-myblock** [comment]
Did you test it thoroughly?

**Comment by CM000n** [comment]
> Did you test it thoroughly?

I checked it against my inverter and with the help of some unit tests. Everything "should" work as before.

**Comment by mawoka-myblock** [comment]
> > Did you test it thoroughly?
> 
> I checked it against my inverter and with the help of some unit tests. Everything "should" work as before.
> 
> ⚠️ **BUT** we are now using type hints folowing [PEP 604](https://peps.python.org/pep-0604/) like `dict | None`. This is only supported with python >= 3.10. So using the package with a python version < 3.10 will now lead to errors. According to the `pyproject.toml` the minimum supported Python version is 3.11. So no problem?

No problem for me.

**Comment by mawoka-myblock** [comment]
I'm gonna take a close look at your awesome work next week, since I'm gonna be busy the next few days.

**Comment by CM000n** [comment]
Hello @mawoka-myblock, have you already had a chance to take a look at the changes? 
I don't want to rush, but I'll probably be out next week and won't have time to make any changes.
So if you have change requests, I could do them this week. Otherwise it would take a while from my side as well.

Best regards and have a relaxed and happy Christmas time for you and your family. 😊 🎄 🎅🏻

**Comment by mawoka-myblock** [comment]
hey, sorry for the late reply, and, merry Christmas 🎄 ! I'm gonna check this week. Sorry again!

**Review by mawoka-myblock — APPROVED** [review]

**Pull request merged**